src/wllama.ts

```diff
@@ -206,6 +206,33 @@ export class Wllama {
     return this.useMultiThread;
   }
 
+  /**
+   * Parses a model URL and returns an array of URLs based on the following patterns:
+   * If the input URL is an array, it returns the array itself.
+   * If the input URL is a string in the `gguf-split` format, it returns an array containing the URL of each shard in ascending order.
+   * Otherwise, it returns an array containing the input URL as a single element array.
+   */
+  parseModelUrl(modelUrl: string | string[]) {
+    if (Array.isArray(modelUrl)) return modelUrl;
+  
+    const urlPartsRegex = /(.*)-(\d{5})-of-(\d{5})\.gguf$/;
```

Comment:
FYI, there is a feature called [Named capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) that we can use here:

```suggestion
    const urlPartsRegex = /(?<baseURL>.*)-(?<current>\d{5})-of-(?<total>\d{5})\.gguf$/;
```